vizro-core/src/vizro/managers/_data_manager.py

```diff
@@ -196,6 +197,40 @@ def __getitem__(self, name: DataSourceName) -> Union[_DynamicData, _StaticData]:
         except KeyError as exc:
             raise KeyError(f"Data source {name} does not exist.") from exc
 
+    def _multi_load(self, multi_name_load_kwargs: list[tuple[DataSourceName, dict[str, Any]]]) -> list[pd.DataFrame]:
```

Comment:
@maxschulz-COL for named data (which includes all dynamic data) this is effectively achieved already. If you want the data for charts to only be loaded once then you use the same data source name and it will only be loaded once.

For unnamed data (which must be static data) it's not really an issue because the data is already in memory and all the `load` operation does is a relatively fast `copy` operation.